ipcore/hdlworkflow.m

```diff
@@ -1,47 +1,39 @@
 %% Load the Model
-model = mp.model_abbreviation;
+model = mp.modelAbbreviation;
 dataplane_name = 'dataplane';
-load_system(model);
-
-if strcmp(mp.target_system,'audiomini')
-    device_family = 'Cyclone V';
-    device = '5CSEBA6U23I7';
-elseif strcmp(mp.target_system,'audioblade')
-    device_family = 'Arria 10';
-    device = '10AS066H2F34I1HG';
-end
 
 %% Model HDL Parameters
 hdlset_param(model, ...
     'DateComment', 'off', ...
     'Workflow', 'Generic ASIC/FPGA', ...
     'ModulePrefix', [model '_'], ...
-    'CriticalPathEstimation', 'off', ...
+    'CriticalPathEstimation', 'on', ...
     'GenerateHDLTestBench', 'off', ...
     'HDLCodingStandardCustomizations',hdlcodingstd.IndustryCustomizations(), ...
     'HDLGenerateWebview', 'off', ...
     'HDLSubsystem', [model '/' dataplane_name], ...
-    'OptimizationReport', 'off', ...
+    'OptimizationReport', 'on', ...
     'ResourceReport', 'off', ...
     'SynthesisTool', 'Altera QUARTUS II', ...
-    'SynthesisToolChipFamily', device_family, ...
-    'SynthesisToolDeviceName', device, ...
+    'SynthesisToolChipFamily', mp.target.deviceFamily, ...
+    'SynthesisToolDeviceName', mp.target.device, ...
     'SynthesisToolPackageName', '', ...
     'SynthesisToolSpeedValue', '', ...
     'MapPipelineDelaysToRAM', 'on', ...
     'UseRisingEdge', 'on', ...
     'TargetDirectory', 'hdlsrc', ...
-    'TargetFrequency', mp.Fs_system);
-
-% TODO: should we always map pipeline delays to ram? 
-% TODO: add ability to customize code generation parameters on a per-model basis 
+    'TargetFrequency', mp.Fs_system / 1000000, ...
+    'Oversampling', 2048, ...
```

Comment:
We can compute the oversampling factor from our model.json config

Reply:
Updated